paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -938,17 +942,6 @@ internal class CustomerSheetViewModel(
 
         return when (val result = intentConfirmationHandler.awaitIntentResult()) {
             is PaymentConfirmationResult.Succeeded -> {
-                safeUpdateSelectPaymentMethodState { viewState ->
```

Comment:
Do we want `refreshAndUpdatePaymentMethods`? If not, why not?

Reply:
Namely, I didn't want to affect the `attach` metrics. A payment method might have been attachd but we failed to refresh the payment methods. Those should be two separate events (hence `CUSTOMER_SHEET_PAYMENT_METHODS_REFRESH_FAILURE`)

Reply:
`refreshAndUpdatePaymentMethods` is still happening but separately from this function. I could move the event reporting though in `handleStripeIntent` and have `refreshAndUpdatePaymentMethods` in there as well.